### PR TITLE
Move configuration to PHP

### PR DESCRIPTION
**Pull request by tuanminhgp**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Part pf #37186
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/esi.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
+use function Symfony\Component\DependencyInjection\Loader\Configurator\service;
+use Symfony\Component\HttpKernel\EventListener\SurrogateListener;
+use Symfony\Component\HttpKernel\HttpCache\Esi;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('esi', Esi::class)
+
```
Comment (misekai) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/esi.php:22]: Please remove the blank line

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/esi.php [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+        ->set('esi_listener', SurrogateListener::class)
+            ->tag('kernel.event_subscriber')
```
Comment (misekai) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/esi.php:24]: please put the `tag` after the `args` for consistency with other pull requests. The original order in the XML file is deprecated.

Comment (tuanminhgp) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/esi.php]: thanks your comment.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/esi.php (hunk 1 above)
Comment (misekai) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/esi.php:22]: Sr my bad. Keep the blank line is clearer! Ignore my comment.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/esi.php [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\Resources\config;
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/esi.php:12]: The namespace should be `namespace Symfony\Component\DependencyInjection\Loader\Configurator;`

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/esi.php [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/esi.php:14]: can be removed with the namespace change

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/esi.php [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+use function Symfony\Component\DependencyInjection\Loader\Configurator\service;
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/esi.php:15]: please remove

**Review by Tobion — CHANGES_REQUESTED** [review]

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Thank you @tuanminhgp.

**Pull request merged**